### ISSUE TITLE
[Session] Fix session hydration considering genesis has block height 1

### ISSUE THREAD

**Issue by red-0ne** [pokt-network/poktroll#377]
## Objective

Correctly hydrate `Session`  by calculating `Session{Start,End}BlockHeight` while taking into consideration that the genesis block has a height of `1` instead of `0`

## Origin Document

The current code assumes that the genesis block has a height of `0` while it is `1`

![image](https://github.com/pokt-network/poktroll/assets/231488/6d33097e-96ae-48e2-9d79-cce77723d738)

## Goals

- Rewrite the functions that are calculating `Session{Start,End}BlockHeight`s considering genesis has the height `1`
- Add a check in `Keeper.GetSession` to verify that the requested height is always `>= 1`
- Rewrite the tests that use height `0` in their assertions
- Make sure all tests are passing

## Deliverables

- [ ] A single PR addressing the genesis block height off-by-one issue.

## General deliverables

- [ ] **Comments**: Add/update TODOs and comments alongside the source code so it is easier to follow.
- [ ] **Testing**: Add new tests (unit and/or E2E) to the test suite.

---

**Creator**: @red-0ne 
**Co-Owners**: @Olshansk

**Comment by Olshansk** [comment]
tl;dr

- `height=0`: `genesis.json` file
- `height=1`: block with `genesis.json` as its parent.

---

To provide additional context here. If we google it, it points to a [dead link](https://hub.cosmos.network/main/resources/genesis.html) showing:

![Screenshot 2024-02-09 at 12 49 02 PM](https://github.com/pokt-network/poktroll/assets/1892194/c1e2d38a-983b-426d-9d51-bc5d3b4187e9)

However, the BNB chain has a [live link](https://docs.bnbchain.org/docs/beaconchain/learn/genesis/) with the same:

![Screenshot 2024-02-09 at 12 50 08 PM](https://github.com/pokt-network/poktroll/assets/1892194/d739c33b-ea1f-431a-8665-8b45e1f9d9c8)

In addition, the [Tendermint Header](https://github.com/tendermint/tendermint/blob/v0.34.x/spec/core/data_structures.md#header) documentation shows:
	
![Screenshot 2024-02-09 at 12 55 18 PM](https://github.com/pokt-network/poktroll/assets/1892194/e7ad5ae2-d029-4f9d-b41d-a2f5e045bfd6)

Since the Cosmos SDK is just a proxy into CometBFT, I went and looked into how Comet comes the header height which led me to [this line](https://github.com/cometbft/cometbft/blob/main/rpc/core/blocks.go#L150) in `cometbft/blob/main/rpc/core/blocks.go`:

```go
// Commit gets block commit at a given height.
// If no height is provided, it will fetch the commit for the latest block.
// More: https://docs.cometbft.com/main/rpc/#/Info/commit
func (env *Environment) Commit(_ *rpctypes.Context, heightPtr *int64) (*ctypes.ResultCommit, error) {
	height, err := env.getHeight(env.BlockStore.Height(), heightPtr)
	if err != nil {
		return nil, err
	}

	blockMeta := env.BlockStore.LoadBlockMeta(height)
	if blockMeta == nil {
		return nil, nil
	}
	header := blockMeta.Header

	// If the next block has not been committed yet,
	// use a non-canonical commit
	if height == env.BlockStore.Height() {
		commit := env.BlockStore.LoadSeenCommit(height)
		return ctypes.NewResultCommit(&header, commit, false), nil
	}

	// Return the canonical commit (comes from the block at height+1)
	commit := env.BlockStore.LoadBlockCommit(height)
	return ctypes.NewResultCommit(&header, commit, true), nil
}
```

The `header` in `NewResultCommit` is what ultimately is passed out to `Cosmos`.
The comment says: `commit for the latest block.`.

If the next block hasn't been committed, they use the last seen commit regardless: `env.BlockStore.LoadSeenCommit(height)`.

In either case, we see that it always has to be a committed block, which is not `0` (i.e. the genesis file itself) and may be `1` using the genesis file as a parent.
